clippy_utils/src/msrvs.rs

```diff
@@ -65,21 +67,15 @@ msrv_aliases! {
     1,15,0 { MAYBE_BOUND_IN_WHERE }
 }
 
-/// Tracks the current MSRV from `clippy.toml`, `Cargo.toml` or set via `#[clippy::msrv]`
-#[derive(Debug, Clone)]
-pub struct Msrv {
-    stack: SmallVec<[RustcVersion; 2]>,
-}
+/// `#[clippy::msrv]` attributes are rarely used outside of Clippy's test suite, as a basic
+/// optimization we can skip traversing the HIR in [`Msrv::meets`] if we never saw an MSRV attribute
+/// during the early lint passes
```

Comment:
Could this break if the user allows all early lints (eg `-Aclippy::all -Wclippy::some_lint`) and all early lint passes get filtered out such that there is nothing to update the static? If I remember right only late lint passes are filtered out, so I guess not (at the moment), but I don't know if that could break in the future?